File: PRACTICAS/practica1/Civitas_P1/src/civitas_p1/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.Random;

/**
 *
 * @author angel
 */
public class Dado {
    
    private Random random;
    private int ultimoResultado;
    private boolean debug;
    
    static final Dado instance = new Dado();
    
    Dado(){
        
        debug = false;
        ultimoResultado = 0;
        random = new Random();
        
    }
    
   static public Dado getInstance() {
    return instance;
  }
   
   int tirar(){
       
       if (!debug)
           ultimoResultado = random.nextInt(6) + 1;
       
       else
           ultimoResultado = 1;
       
       return ultimoResultado;
       
       
   }
   
   boolean salgoDeLaCarcel(){
       return tirar() >= 5;
   }
   
   int quienEmpieza(int n){
       return random.nextInt(n);
   }
   
   void setDebug(boolean d){
       
       debug = d;
       
       if (d)
           Diario.instance.ocurreEvento("El debug se ha puesto a true");
       else
           Diario.instance.ocurreEvento("El debug se ha puesto a false");
   }
   
   int getUltimoResultado(){
       return ultimoResultado;
   }   
    
}

File: PRACTICAS/practica3/Civitas_Ruby/pkg/Civitas_Ruby-0.0.1/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require 'singleton'

module Civitas
  
  class Dado
    include Singleton
    
    @@SalidaCarcel = 5
    
    private
    def initialize
      @random = rand(6) + 1
      @ultimo_resultado = 0
      @debug = false
    end
    
    public
    
    attr_reader :ultimo_resultado
    
    def tirar  
      
      if @debug       
        @ultimo_resultado=1       
      else   
        @ultimo_resultado = rand(6) + 1        
      end
      
      return @ultimo_resultado
      
    end
    
    def salgo_de_la_carcel      
      return tirar >= @@SalidaCarcel    
    end
  
    def quien_empieza(num_jugadores)     
      return rand(num_jugadores)              
    end
  
    def set_debug(d)
      
      @debug = d
      
      if @debug       
        Diario.instance.ocurre_evento("El debug esta activado")         
      else   
        Diario.instance.ocurre_evento("El debug esta desactivado")     
      end
      
    end
   
    
  end
  
end